[trpmy] iP

trpmy opened the pull request:
### Langley ❈

> "He's been waiting for so long." - Sunless Skies

- Langley can greet you
- and can send you off.

Tell him what you need to do:
1. ~Todo~
2. **Deadline**
3. _Event_

He can, and will:

- [x] Be here
- [x] Be there
- [x] Be in the `Langley` class
- [x] And will [not give you up](https://www.youtube.com/watch/dQw4w9WgXcQ)

Find Langley where you first meet him:

```java
package langley;

import langley.*;

public class Langley {
    public static void main(String[] args) {
        // Langley doing Langley stuff
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Named bot
Changed greeting
Added goodbye

trpmy committed:
Added echo functionality

trpmy committed:
Added list history

trpmy committed:
Added mark toggle per message
list now lists with mark toggles

trpmy committed:
Added support for Todo, Deadline, Event

trpmy committed:
Error messages standardised and clearer

trpmy committed:
Added delete functionality

trpmy committed:
Automate Text UI Testing

trpmy committed:
Add save functionality

trpmy committed:
Add date comprehension to timed tasks

trpmy committed:
Refactor code for better OOP adherence

trpmy committed:
Refactor into packages

trpmy committed:
Add header comments and specify imports

trpmy committed:
Merge remote-tracking branch 'origin/add-gradle-support'

S-K-Y-Light commented on src/main/java/langley/tasks/Task.java:
```diff
@@ -0,0 +1,75 @@
+
+public abstract class Task {
+    String description;
+    boolean isMarked;
```
It's a little unclear what "isMarked" stands for. Maybe changing it to "isDone" would be clearer?

S-K-Y-Light commented on src/main/java/langley/tasks/Task.java:
```diff
@@ -0,0 +1,75 @@
+                task = new Event(eventDesc, from, to);
+                break;
+            default:
+                return null;
```
Indentation for Switch Cases violates CS2103/T Coding Standards.

LiHaixin000 commented on src/main/java/langley/Langley.java:
```diff
@@ -0,0 +1,101 @@
+     *
+     * @param args Command-line arguments (not used).
+     */
+    public static void main(String[] args) {
```
The main method is quite long. Perhaps some functionality be extracted into separate methods to improve readability and maintainability?

S-K-Y-Light commented on src/main/java/langley/tasks/Task.java:
```diff
@@ -0,0 +1,75 @@
+                break;
+            case "D":
+                String[] deadlineParts = description.split("\\(by: ");
+                String deadlineDesc = deadlineParts[0].trim();
```
Using "Desc" might be unclear. Maybe changing to "Description" would improve clarity?
Similar Issue for "eventDesc" in Task.java.

S-K-Y-Light commented on src/main/java/langley/tasks/Task.java:
```diff
@@ -0,0 +1,75 @@
+     * @param line The string representing a task.
+     * @return The corresponding Task object or null if parsing fails.
+     */
+    public static Task fromString(String line) {
```
I think "fromString" is rather vague and could be renamed to something more representative of what the method does.
Maybe something like "createTaskFromString" would be clearer?

S-K-Y-Light commented on src/main/java/langley/ui/Ui.java:
Since Ui has a "print" method defined, maybe the other methods can make use of "print" instead of System.out.println()?

LiHaixin000 commented on src/main/java/langley/tasks/Task.java:
```diff
@@ -0,0 +1,75 @@
+
+        Task task = null;
+
+        switch (type) {
```
The switch statement should have the following form: Note there is no indentation for case clauses.

```java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
} 
```

S-K-Y-Light commented on src/main/java/langley/storage/Storage.java:
There are extra Line Breaks between the Method Documentation and the methods.
Maybe remove the line breaks to keep consistency with other files' documentations?

S-K-Y-Light reviewed (commented):
Just some minor issues with names, documentation & coding standard. Otherwise, I like your code. LGTM.

LiHaixin000 commented on src/main/java/langley/tasks/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
```
Maybe writing descriptive header comments for all public classes/methods could improve the readability of the code?
The following suggestions are extracted from [here](https://se-education.org/guides/conventions/java/intermediate.html)
You MUST write header comments for all classes, public methods. But they can be omitted for the following cases:
- Getters/setters
- When overriding methods (provided the parent method's Javadoc applies exactly as is to the overridden method)

LiHaixin000 commented on src/main/java/langley/storage/Storage.java:
```diff
@@ -0,0 +1,62 @@
+import java.io.FileReader;
+import java.util.ArrayList;
+
+public class Storage {
```
I noticed the missing Javadoc in several other places too. Consider adding them to make your code even better

LiHaixin000 reviewed (commented):
LGTM. Just some nits to fix

trpmy committed:
Support JUnit testing for TaskList, Task

trpmy committed:
Add class header comments

trpmy committed:
Standardise import and switch indentation

trpmy committed:
Merge branch 'branch-A-CodingStandard'

trpmy committed:
Add search functionality, restrict some access modifiers

trpmy committed:
Merge branch 'branch-Level-9'

trpmy committed:
Add gradle dependencies for JavaFX

trpmy committed:
Add GUI

This adds a basic GUI to improve UX.

CLI input is no longer supported. The Ui class is obsolete.

This shouldn't change UX for most users.
Text commands still function as per normal,
only now there's a GUI.

trpmy committed:
Add assert to validate inputs

trpmy committed:
Improve code quality

trpmy committed:
Merge pull request #2 from trpmy/branch-A-Assertions

Add assert to validate inputs

trpmy committed:
Merge pull request #3 from trpmy/branch-A-CodeQuality

Improve code quality

trpmy committed:
Improve find functionality

Now all searching is case-insensitive.
Previously, search was case-sensitive.

trpmy committed:
Merge remote-tracking branch 'origin/master'

trpmy committed:
Add screenshot of app interface

trpmy committed:
Update README.md
